### PR TITLE
[Revised] Update docs

### PR DESCRIPTION
**Pull request by bridadan**
This is an update of #214, but the `CONTRIBUTING` doc has been left in. We should create a new task to update this doc for the future.

@mazimkhan @AnotherButler Could you please take a look?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -563,196 +96,87 @@ In the below example there are two builds defined:
...
-will pick up test specification and execute all tests in it.
+## Command line usage
+Here we will highlight a few of the capabilities of the Greentea command line interface. For a full list of the availble options, please run `mbedgt --help`.
 
-#### Cherry-pick tests
+We will assume for the examples below that the above `test_spec.json` file is in current directory.
 
-* We will first list the tests we want to execute:
+### Listing all tests
+We can use the `-l` arguement to list all availble tests:
```
Comment (adbridge) [inline README.md]: Some typos - arguement and availble

File: README.md [hunk 2]
```diff
@@ -563,196 +96,87 @@ In the below example there are two builds defined:
...
 $ mbedgt -l
-```
-```
 mbedgt: using 'test_spec.json' from current directory!
-mbedgt: available tests for built 'K64F-ARM', location './.build/K64F/ARM'
-        test 'mbed-drivers-test-generic_tests'
-        test 'mbed-drivers-test-c_strings'
-mbedgt: available tests for built 'K64F-GCC', location './.build/K64F/GCC_ARM'
-        test 'mbed-drivers-test-generic_tests'
+mbedgt: available tests for built 'K64F-ARM', location './BUILD/K64F/ARM'
```
Comment (adbridge) [inline README.md]: shouldn't 'built' be 'build' ?

File: README.md [hunk 3]
```diff
@@ -765,92 +189,39 @@ $ mbedls
...
-  c:\python27\scripts\mbedgt.exe
-  c:\python27\scripts\mbedgt.exe.manifest
-Proceed (y/n)? Y
-  Successfully uninstalled mbed-greentea
+mbedgt --report-text text_report.text
 ```
 
-# Commissioning mbed platforms
+## Host test detection
+When developing with mbed OS, Grentea will detect host tests automatically if they placed in the correct location. All tests in mbed OS are placed under a `TESTS` directory. You may place custom host test scripts in a folder named `host_tests` in this folder. For more information on the mbed OS test directory structure, please see the [mbed CLI documentation](https://docs.mbed.com/docs/mbed-os-handbook/en/latest/dev_tools/cli/#test-directory-structure)
```
Comment (adbridge) [inline README.md]: "if they are placed in the correct location"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,9 +1,203 @@
...
+
+## Workflow
+### Forking and cloning this repository
+First [fork](https://help.github.com/articles/fork-a-repo/) this repository in GitHub, then clone it locally with:
+```
+$ git clone <repo-link>
+```
+Now you can create separate branches in the forked repository and prepare pull requests with changes.
+### Issues solving
+Simple work-flow issue solving process may contain below steps:
```
Comment (adbridge) [inline CONTRIBUTING.md]: A simple work-flow for solving issues may follow the steps below:

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,9 +1,203 @@
...
+$ git commit -m "Add fix for issue #38" -m "More verbose explanation of the change/fix"
+$ git push origin issue_38
+
+```
+
+* Push changes to GitHub.
+* Create pull request from GitHub webpage (your fork's dashboard).
+
+### Branch naming conventions
+We prefer is you use standardised naming convention when creating pull requests.
```
Comment (adbridge) [inline CONTRIBUTING.md]: We prefer you to use a standardised ...

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,9 +1,203 @@
...
+Below few example of branch names' prefixes you could use when creating pull request from your fork:
```
Comment (adbridge) [inline CONTRIBUTING.md]: Here are a few examples of branch name prefixes...

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,9 +1,203 @@
...
+* ```issue_``` - branch with fix for issue. E.g. ```issue_38```.
+* ```doc_``` - documentation update. E.g. ```doc_add_faq```.
+* ```devel_``` - development of a new feature. E.g. ```devel_udp_client_test```.
+* ```test_``` - when pull request will consist of only new/updates to test cases. E.g. ```test_paralllel_execution```.
+
+## Coding style and coding rules
+This chapter attempts to explain the basic styles and patterns that are used in mbed test tools projects. he following norms should be followed for new code, and for code that needs clean-up.
```
Comment (adbridge) [inline CONTRIBUTING.md]: The following rules should be followed for producing new code and updating existing code.

File: README.md [hunk 8]
```diff
@@ -94,15 +94,15 @@ In the below example there are two builds defined:
 }
 ```
 
-In below examples we will use the above test specification file.
+The examples below use the above test specification file.
 
-## Command line usage
-Here we will highlight a few of the capabilities of the Greentea command line interface. For a full list of the availble options, please run `mbedgt --help`.
+## Command-line usage
+This section highlights a few of the capabilities of the Greentea command-line interface. For a full list of the availble options, please run `mbedgt --help`.
```
Comment (adbridge) [inline README.md]: availble ?

**Comment by bridadan** [comment]
@adbridge I've addressed your comments, however please note the CONTRIBUTING doc has other issues in it. For this reason I suggest we come back to that doc a later date.

**Pull request merged**